src/client/CustomerClientApi.ts

```diff
@@ -0,0 +1,34 @@
+import { Customer, CustomersApi, SearchCustomersRequest } from 'square-connect';
+import { IFindOrCreateCustomerRequest } from '../interface';
+
+export class CustomerClientApi extends CustomersApi {
+    async findOrCreateCustomer(findOrCreateCustomerRequest: IFindOrCreateCustomerRequest): Promise<Customer> {
+        const { phone_number }: IFindOrCreateCustomerRequest = findOrCreateCustomerRequest;
+
+        if (!phone_number) {
+            throw Error('Not found phone number');
+        }
+
+        const searchCustomerReq: SearchCustomersRequest = {
+            query: {
+                filter: {
+                    phone_number: {
+                        exact: phone_number,
+                    },
+                },
+            },
+        };
+
+        const { customers } = await this.searchCustomers(searchCustomerReq);
```

Comment:
we are not allowed to return a customer just by phone number - only full match by phone number & email.
If email does not match just create new payment customer